embedded/sql/row_reader.go

```diff
@@ -67,6 +68,31 @@ func (row *Row) Compatible(aRow *Row, selectors []*ColSelector, db, table string
 	return true, nil
 }
 
+func (row *Row) digest() (d [sha256.Size]byte, err error) {
+	h := sha256.New()
+
+	for _, v := range row.Values {
```

Comment:
The iteration order is non-deterministic for hash maps, this may have unexpected results.

I checked and even reserving map size changes the order:
https://play.golang.org/p/LZMrO6a5n-W